### PR TITLE
Updating dependency versions

### PR DESCRIPTION
**Pull request by pokearu**
*Description of changes:*
Bumping cloudstack to latest and changing the linuxkit golang version.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: projects/apache/cloudstack-cloudmonkey/Makefile [hunk 1]
```diff
@@ -1,6 +1,6 @@
 BASE_DIRECTORY:=$(shell git rev-parse --show-toplevel)
 GIT_TAG:=$(shell cat GIT_TAG)
-GOLANG_VERSION?="1.16"
+GOLANG_VERSION?="1.20"
```
Comment (jaxesn) [inline projects/apache/cloudstack-cloudmonkey/Makefile:3]: are they uysing this version upstream?

File: projects/tinkerbell/hook/Makefile [hunk 2]
```diff
@@ -52,7 +52,7 @@ s3-artifacts: $(CREATE_HOOK_FILES)
 $(CREATE_HOOK_FILES): tarballs
 # Modify the linuxkit config file hook/hook.yaml to point to IMAGE_REPO.
 # Upstream make target `dist` is triggered to perform linuxkit build and generate OSIE files ().
-	source $(BUILD_LIB)/common.sh && build::common::use_go_version "1.16" && make dist IMAGE_REPO=$(IMAGE_REPO) LATEST_TAG=$(LATEST_TAG) -C $(REPO)
+	source $(BUILD_LIB)/common.sh && build::common::use_go_version "1.19" && make dist IMAGE_REPO=$(IMAGE_REPO) LATEST_TAG=$(LATEST_TAG) -C $(REPO)
```
Comment (jaxesn) [inline projects/tinkerbell/hook/Makefile:55]: do we need to update linuxkit in the builder base? if not, can we build it with a newer golang?

File: projects/apache/cloudstack-cloudmonkey/Makefile (hunk 1 above)
Comment (abhay-krishna) [reply projects/apache/cloudstack-cloudmonkey/Makefile:3]: Yes they released a new tag in April after 1.5 years. They jumped from 1.14 to 1.20 in their CI workflows.

File: projects/tinkerbell/hook/Makefile (hunk 2 above)
Comment (pokearu) [reply projects/tinkerbell/hook/Makefile:55]: Yup we updated the version to what upstream hook is tracking.

**Pull request merged**